core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java

```diff
@@ -114,16 +126,61 @@ private static void handleObject(String prefix, Object o, SmallRyeConfig config)
         }
     }
 
-    private static Converter<?> getConverterFor(Type type) {
+    private static Map<?, ?> handleMap(String fullName, Type genericType, SmallRyeConfig config,
+            List<String> quarkusPropertyNames) throws ReflectiveOperationException {
+        var map = new HashMap<>();
+        if (typeOfParameter(genericType, 0) != String.class) { // only support String keys
+            return map;
+        }
+        var processedSegments = new HashSet<String>();
+        // infer the map keys from existing property names
```

Comment:
So I'm extrating the segments/map keys form all available property names.

At this point I have no idea whether or not this covers env var flavors (e.g. `QUARKUS_FOO`).